File: tests/fixtures/php/deep_nesting.php
<?php

function deeply_nested(int $a, int $b, int $c, int $d, int $e): int {
    if ($a > 0) {
        if ($b > 0) {
            for ($i = 0; $i < $c; $i++) {
                if ($d > 0) {
                    while ($e > 0) {
                        $e--;
                    }
                }
            }
        }
    }
    return $e;
}

File: tests/fixtures/lua/deep_nesting.lua
function deeplyNested(items, threshold)
    local count = 0
    for _, item in ipairs(items) do
        if item > threshold then
            if item < 200 then
                if item ~= 128 then
                    if count < 1000 then
                        count = count + 1
                    end
                end
            end
        end
    end
    return count
end

function moderatelyNested(x, y)
    if x > 0 then
        if y > 0 then
            return x + y
        end
    end
    return 0
end
